### PR TITLE
Keyboard fixes and new shortcuts

### LINKED ISSUE #51: Taranis also reacts on keyboard presses intended for the browser
- Got to asses where some news item is selected
- Press some keyboard combination intended for the browser, e.g. Ctrl+Shift+I in Firefox to open the developer console
- The news item is marked as important

### PR DESCRIPTION
**Pull request by sebix**
This adds new functionality and fixes some bugs of the keyboard shortcuts but some remain.

- keyboard shortcuts: add jk, Enter, Esc for news item navigation
in addition to the arrow keys, this allows to use the jk keys to be
used to navigate through the news items.
These keys are widely used for navigation and allow the fingers to rest
at their natural position on the keyboard.
Using the Escape to exit the news item detail view is intuitive
and using enter to open the item as well.
-    gui keyboard mixins: remove unused _keyAction
gui: keyboard: match key presses more accurately
- use the characters themselves, allows differentiation between
lower/upper case
check if Ctrl and Alt are not pressed to prevent catching other events
which are not meant to be treated (shortcuts handled by the browser)
fixes #51
- new keyboard shortcut to open news item source
'o' opens the news item source of the selected or opened news item
does not open multiple selected news items though
-  gui: keyboard: move preventDefault to section where it fits better
does only apply to a minority of cases
- gui keyboard: allow shortcuts when in focus
some shortcuts are useful when an item is in focus, so move the
detection mechanismis outside the focus-clause
- gui shortcuts for search function
/ to focus the search bar
When in the search bar:
Esc clears the text
With another Esc if the field is empty un-sets the focus, allows the
other shortcuts to work

### CODE REVIEW

**Comment by sebix** [comment]
New: shortcuts for the search bar to enter and escape from it

**Comment by sebix** [comment]
New features:
- arrow up/dow or j/k scrolls the news item text which is visible
- jump to the first/last news item with Home/End keys
- jump to reasonable news item after marking news items
after marking multiple news items as read/important/liked/delete it,
set the focus to the next news item in order if the active filter will
hide it then.
Otherwise the cursor would be at a position below the last news item
selected

**Comment by sebix** [comment]
Rebased on main.

**Comment by sebix** [comment]
based on #56 , so please merge the other PR first

File: src/gui/src/assets/keyboard_mixin.js [hunk 1]
```diff
@@ -83,12 +83,12 @@ const keyboardMixin = targetId => ({
             card.show = this.card_items[this.pos].querySelector(".card");
             card.id = this.card_items[this.pos].dataset.id;
             card.close = document.querySelector("[data-dialog='" + dialog + "-detail'] [data-btn='close']");
-            //card.close = document.querySelector(".v-dialog--active [data-btn='close']");
+            // Link is always extracted from the news item newdial button, does not work for opening all selected items, but also with opened items
+            card.link = this.card_items[this.pos].querySelector(".newdial button[data-btn='link'] a");
```
Comment (milankowww) [inline src/gui/src/assets/keyboard_mixin.js:87]: this refers to an obsolete .newdial class and probably wouldn't work

**Review by milankowww — APPROVED** [review]
There is one place in keyboard_mixin.js where `newdial` was used. We will fix by a separate commit after merge.

Thanks for the work!

File: src/gui/src/assets/keyboard_mixin.js (hunk 1 above)
Comment (sebix) [reply src/gui/src/assets/keyboard_mixin.js:87]: I have some more commits locally fixing that, will create a new PR with some stuff.

**Pull request merged**